lib/handler/configurator/proxy.c

```diff
@@ -618,10 +618,23 @@ static int on_config_http3_ratio(h2o_configurator_command_t *cmd, h2o_configurat
 static int on_config_expect(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
 {
     struct proxy_configurator_t *self = (void *)cmd->configurator;
-    ssize_t ret = h2o_configurator_get_one_of(cmd, node, "OFF,ON");
+    ssize_t ret = h2o_configurator_get_one_of(cmd, node, "OFF,ON,FORWARD");
     if (ret == -1)
         return -1;
-    self->vars->conf.use_expect = (int)ret;
+    switch (ret) {
+    case 0:
+        self->vars->conf.expect_mode = H2O_PROXY_EXPECT_DISABLED;
+        break;
+    case 1:
+        self->vars->conf.expect_mode = H2O_PROXY_EXPECT_ENABLED;
+        break;
+    case 2:
+        self->vars->conf.expect_mode = H2O_PROXY_EXPECT_FORWARD;
+        break;
+    default:
+        return -1;
```

Comment:
`h2o_fatal` would be a better choice here. We do not want to see the program running in an unexpected state after a logic flaw.

Reply:
hmm, but it seems not aligned with the other parts of this file, e.g. `on_config_zerocopy`. Do we have to tweak it as well then?

Reply:
In the two functions that you point out, the path that returns -1 is taken when `h2o_configurator_get_one_of` returns -1 (i.e., when the value in the configuration file does not match any of the provided values).

In the case of the code added in this PR, that check is done before the `switch` statement in question. Therefore this `default` label and the return is excessive.

Reply:
ah! I see the point. thank you